docs/SECURITY_CONSIDERATIONS.md

```diff
@@ -19,6 +19,52 @@ assertion.repository_owner == 'my-github-org'
 Never use a "*" in an IAM Binding unless you absolutely know what you are doing!
 
 
+## Set up Repository & Branch Protection
+
+An attribute condition checking just the `repository_owner` assertion is most likely insufficient. Since your workflow may access secrets and other sensitive cloud resources, it should only be run on trusted code. Untrusted code can come from other repositories in the same organization, or even unreviewed branches in the same repository. The only code that should be trusted is code that has been reviewed and merged into a protected main branch.
+
+To restrict access to a workflow on your main branch, you'll need to do the following:
+
+1. Map the necessary GitHub assertions to Google STS token attributes.
+2. Create attribute conditions that are verified whenever a GitHub OIDC token is exchanged for a Google STS token.
+
+### Mapping GitHub Assertions to Google STS Token Attributes
+
+You can use the following mappings to provide Google with enough information to verify the GitHub OIDC token.
+
+```
+attribute.repository = assertion.repository
+attribute.event_name = assertion.event_name
+attribute.base_ref   = assertion.base_ref
+attribute.ref        = assertion.ref
+attribute.workflow   = assertion.workflow
+```
+
+More information about each claim from the GitHub OIDC token can be found [here](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#configuring-the-oidc-trust-with-the-cloud).
+
+### Create Attribute Conditions for Protected Main Branch
+
+Once the GitHub assertions have been mapped to Google STS token attributes, you can create attribute conditions to verify that the caller is coming from a trusted source. For example, the following conditions verify that the GitHub OIDC token was generated after a push to the `main` branch of the `octo-org/octo-repo` repository.
+
+```
+attribute.repository == "octo-org/octo-repo" && \
+  attribute.event_name == "push" && \
+  attribute.ref == "refs/heads/main"
+```
+
+You can also use the `workflow` attribute if you want to restrict access to specific workflows.
+
+### Create Attribute Conditions for Pull Requests
```

Comment:
Suggestion: Have one more section for creating attributes [based on Github Environments](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#filtering-for-a-specific-environment).

Environments are a Github Enterprise feature, so not everyone may have access, but using environments make it marginally easier to categorize branches within an attribute. For example in a github action workflow you can set an if conditional like so:

`if: github.ref_protected == true && github.event_name == 'push'`

Which you'll note encapsulates all protected branches (not just main). If the repo only has a single protected repository, it's equivalent to your section on [Attribute Conditions for Protected Main Branch](https://github.com/google-github-actions/auth/pull/409/files#diff-60237a245d9cea438b88ace2a546f7cbc376851714d3d337f5befdd5907ce9e9R45), but in your section you have to individually set `attribute.ref` to the branch. Whereas if you use the `sub` claim of the attribute which references a Github Environment, you can have the environment be a superset of all protected branches. And would look something like this:

`attribute.sub == "repo:octo-org/octo-repo:environment:Production"`

Reply:
One thing I'm not 100% sure about is why we need to add the `if` condition to the workflow. The claim verification is done on the cloud provider's side, right? Even if random, unprotected branches triggered the workflow, they wouldn't be able to access the cloud provider's resources.